File: data/src/main/java/top/contins/synapse/data/repository/TaskRepository.java
package top.contins.synapse.data.repository;

import androidx.lifecycle.LiveData;
import top.contins.synapse.domain.entity.Task;
import java.util.List;

/**
 * 任务数据仓库接口
 */
public interface TaskRepository {
    LiveData<List<Task>> getAllTasks();
    LiveData<Task> getTaskById(String id);
    void insertTask(Task task);
    void updateTask(Task task);
    void deleteTask(String taskId);
    LiveData<List<Task>> getTasksByStatus(top.contins.synapse.domain.entity.TaskStatus status);
}

File: domain/src/main/kotlin/top/contins/synapse/domain/repository/TaskRepository.kt
package top.contins.synapse.domain.repository

import top.contins.synapse.domain.model.Task
import kotlinx.coroutines.flow.Flow

/**
 * 任务数据仓库接口
 */
interface TaskRepository {
    fun getAllTasks(): Flow<List<Task>>
    suspend fun getTaskById(id: String): Task?
    suspend fun createTask(task: Task)
    suspend fun updateTask(task: Task)
    suspend fun deleteTask(id: String)
}
